### PR TITLE
DTSPO-6998 - build testapp image via pipeline

### PR DESCRIPTION
**Pull request by msl8r**
Build and push test app image automatically

https://tools.hmcts.net/jira/browse/DTSPO-6998

### CODE REVIEW

File: azure-pipelines.yaml [hunk 1]
```diff
@@ -3,50 +3,82 @@ name: Nodejs Chart CI Pipeline
...
   repositories:
-  - repository: cnp-library
-    type: github
-    name: hmcts/cnp-azuredevops-libraries
-    endpoint: hmcts
+    - repository: cnp-library
+      type: github
+      name: hmcts/cnp-azuredevops-libraries
+      endpoint: hmcts
+      ref: msl8r-patch-1
```
Comment (timja) [inline azure-pipelines.yaml:19]: :shipit:

Comment (msl8r) [reply azure-pipelines.yaml:19]: not ready yet :)

File: azure-pipelines.yaml [hunk 2]
```diff
@@ -3,50 +3,76 @@ name: Nodejs Chart CI Pipeline
...
+              buildPath: $(buildPath)
+
+      - template: steps/charts/validate.yaml@cnp-library
+        parameters:
+          chartNamespace: chart-tests
+          chartName: nodejs
+          chartReleaseName: chart-nodejs-ci-test
+          chartPath: $(repoName)/
+          valuesFile: $(valuesFile)
+          additionalHelmArgs: --set image=$(testAppImage):pr-$(System.PullRequest.PullRequestNumber)
```
Comment (timja) [inline azure-pipelines.yaml:59]: isn't this always adding a PR number? even on master?

Comment (msl8r) [reply azure-pipelines.yaml]: yep, fixed

Comment (msl8r) [reply azure-pipelines.yaml]: maybe not...

Comment (msl8r) [reply azure-pipelines.yaml]: no it is fixed

File: azure-pipelines-image-publish.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+name: Nodejs testapp image build and push Pipeline
```
Comment (timja) [inline azure-pipelines-image-publish.yaml:1]:
```suggestion
name: chart-nodejs test-app
```

**Review by timja — APPROVED** [review]

**Pull request merged**